display_list/display_list_utils.cc

```diff
@@ -152,6 +152,11 @@ void SkMatrixDispatchHelper::transformFullPerspective(
 
 // clang-format on
 
+void SkMatrixDispatchHelper::transformReset() {
+  matrix_ = {};
+  matrix33_ = {};
```

Comment:
Hmm, I think saved_ needs to be cleared as well. And, I think there is a bug here where calling an imbalaned restore will call pop_back on an empty vector which is undefined behavior.

Fixing both issues.

Reply:
Nevermind. I can't confirm this.

Reply:
SkMatrixDispatchHelper already has a reset() method. I'm not sure if it is used any more (I used to collect saveLayer bounds in their native coordinate space, but had to back off on that due to bugs).

You don't want saved_ to be cleared, if they restore() then they should undo the reset. The set to identity should not escape the enclosing save/restore.

There probably should be a protection here for over-restore, but the caller generally already does that. It would be redundant for existing use cases, but we shouldn't rely on that.

Reply:
Filed https://github.com/flutter/flutter/issues/100176

Reply:
Wait, do these set to identity? Or to all zeros?